### PR TITLE
Fix vectorize size calculation

### LINKED ISSUE #1843: TIMM Stride Vectorization Error
### 🐛 Describe the bug

Fusion fails with fallback path with channels_last + amp + batch_size > 1

Steps to reproduce:
`PYTORCH_NVFUSER_DISABLE_FALLBACK=1 python benchmark.py --bench train --model rexnet_100 --img-size 224 -b 2 --amp --torchscript --channels-last --no-retry`

Failing networks:
- rexnet_100
- rexnet_130
- rexnet_150
- rexnet_200

General Error:

> RuntimeError: `stride == cur_contig_stride || size == 1 || (still_rightmost && stride == 1) || (!still_rightmost && stride % word_size == 0)` 
> INTERNAL ASSERT FAILED at "../torch/csrc/jit/codegen/cuda/executor_utils.cpp":591, please report a bug to PyTorch. 
> `Vectorization of T2_g[ iS128{( ceilDiv(( ceilDiv(( ceilDiv(( T0.size[0] * ( T0.size[1] * ( T0.size[2] * T0.size[3] ) ) ), 4) ), 1) ), 128) )}, iS127{1}, iS125{4}, iS129{128} ] with word size **XXX** not possible due to invalid stride. Domain: iS125{4}, stride: **XXX**`

### Versions

1. nvfuser devel branch

### PR DESCRIPTION
**Pull request by zasdfgbnm**
Fixes #1843

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
This is just a prototype now, it should work with view, but doesn't work with transpose/reduction/normalization.

File: torch/csrc/jit/codegen/cuda/test/test_gpu.cpp [hunk 1]
```diff
@@ -26125,6 +26125,160 @@ TEST_F(NVFuserTest, FusionTrivialInputForwarding_CUDA) {
   testValidate(fusion, cg_outputs2, {t0, t1}, {t0}, __LINE__, __FILE__);
 }
 
+namespace {
```
Comment (zasdfgbnm) [inline torch/csrc/jit/codegen/cuda/test/test_gpu.cpp:26128]: We should split this file. It is too long that GitHub no longer displays syntax highlights.

File: torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.cpp [hunk 2]
```diff
@@ -202,8 +190,16 @@ size_t expandVectorizationToContigMergedDomains(
 
   // Merge the domains right of the break point
   const auto& ref_root = reference_tv->getMaybeRFactorDomain();
-  const int num_merged_domains =
+  const int max_num_merged_domains =
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.cpp:193]: I'm going to have to relook at how I reimplemented this. I think it should be safe, but will have to pull in the new tests and check obviously.

Relevant changes I made:
https://github.com/csarofeen/pytorch/pull/2039/files#diff-cb1f12c937ff579f33d244faf2152ecf9a4a28a8ca7203863bcf158de149f8eeR287

**Review by csarofeen — APPROVED** [review]
LGTM Thanks!

File: torch/csrc/jit/codegen/cuda/test/test_gpu.cpp (hunk 1 above)
Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/test/test_gpu.cpp:26128]: Is anyone against with just naming test files like `test_gpu1.cpp`, `test_gpu2.cpp`, etc? We should continue have topic-specific test files like `test_gpu_view.cpp`, but there are many that don't belong to one specific category. An easy way is to rename `test_gpu.cpp` to `test_gpu1.cpp` and don't add new tests there anymore. New tests should go into `test_gpu2.cpp` until that file becomes too large, then `test_gpu3.cpp` should be created. Not super neat, but I think it should be just enough.

@zasdfgbnm @csarofeen

**Pull request merged**